### PR TITLE
[docs-only] Add atomicity docs for decomposedfs operations

### PR DESCRIPTION
**Pull request by aduffeck**
This pr adds a document describing the atomicity of the writing decomposedfs operations.

### CODE REVIEW

File: pkg/storage/utils/decomposedfs/atomicity.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+### Potential Problems
+
+Several concurrent `CreateDir` calls can get past the exit critera step 1 because the directory does not exist yet.
+Each of the calls generates a new ID, creates the according node on disk and tries to link it to the parent.
+Only the first one succeeds in that, the later ones fail because the link already exists (See
+`Considerations > Creating symlinks`).
+
+### Negative Effects
+
+Failing calls will leave an orphened node behind (See reva issue [#1601](https://github.com/cs3org/reva/issues/1601)).
```
Comment (ishank011) [inline pkg/storage/utils/decomposedfs/atomicity.md:24]: Typo

**Review by ishank011 — COMMENTED** [review]
Hi @aduffeck this looks good! Would it make more sense to add this to https://github.com/cs3org/reva/tree/master/docs/content/en/docs/config/packages/storage/fs? That way it would be present in the config docs as well (similar to https://reva.link/docs/config/http/services/mentix/)

**Comment by aduffeck** [comment]
@ishank011 I fixed the typo and moved the docs as suggested.

**Review by ishank011 — APPROVED** [review]
https://deploy-preview-1645--cs3org-reva.netlify.app/docs/config/packages/storage/utils/decomposedfs/atomicity/ 🚀

**Pull request merged**